pydantic/_internal/_std_types_schema.py

```diff
@@ -219,71 +261,92 @@ def ordered_dict_schema(schema_generator: GenerateSchema, obj: Any) -> core_sche
         js_metadata=JsonSchemaMetadata(source_class=IPv4Network, type='string', format='ipv4network')
     )
-    return core_schema.function_plain_schema(
-        _validators.ip_v4_network_validator, serialization=core_schema.to_string_ser_schema(), metadata=metadata
+    return core_schema.lax_or_strict_schema(
+        lax_schema=core_schema.function_plain_schema(_validators.ip_v4_network_validator, metadata=metadata),
+        strict_schema=core_schema.is_instance_schema(IPv4Network),
+        serialization=core_schema.to_string_ser_schema(),
     )
 
 
 @schema_function(IPv6Address)
-def ip_v6_address_schema(_schema_generator: GenerateSchema, _obj: Any) -> core_schema.FunctionPlainSchema:
+def ip_v6_address_schema(_schema_generator: GenerateSchema, _obj: Any) -> core_schema.LaxOrStrictSchema:
     metadata = build_metadata_dict(
         js_metadata=JsonSchemaMetadata(source_class=IPv6Address, type='string', format='ipv6')
     )
-    return core_schema.function_plain_schema(
-        _validators.ip_v6_address_validator, serialization=core_schema.to_string_ser_schema(), metadata=metadata
+    return core_schema.lax_or_strict_schema(
+        lax_schema=core_schema.function_plain_schema(_validators.ip_v6_address_validator, metadata=metadata),
+        strict_schema=core_schema.is_instance_schema(IPv6Address),
+        serialization=core_schema.to_string_ser_schema(),
     )
 
 
 @schema_function(IPv6Interface)
-def ip_v6_interface_schema(_schema_generator: GenerateSchema, _obj: Any) -> core_schema.FunctionPlainSchema:
+def ip_v6_interface_schema(_schema_generator: GenerateSchema, _obj: Any) -> core_schema.LaxOrStrictSchema:
     metadata = build_metadata_dict(
         js_metadata=JsonSchemaMetadata(source_class=IPv6Interface, type='string', format='ipv6interface')
     )
-    return core_schema.function_plain_schema(
-        _validators.ip_v6_interface_validator, serialization=core_schema.to_string_ser_schema(), metadata=metadata
+    return core_schema.lax_or_strict_schema(
+        lax_schema=core_schema.function_plain_schema(_validators.ip_v6_interface_validator, metadata=metadata),
+        strict_schema=core_schema.is_instance_schema(IPv6Interface),
+        serialization=core_schema.to_string_ser_schema(),
     )
 
 
 @schema_function(IPv6Network)
-def ip_v6_network_schema(_schema_generator: GenerateSchema, _obj: Any) -> core_schema.FunctionPlainSchema:
+def ip_v6_network_schema(_schema_generator: GenerateSchema, _obj: Any) -> core_schema.LaxOrStrictSchema:
     metadata = build_metadata_dict(
         js_metadata=JsonSchemaMetadata(source_class=IPv6Network, type='string', format='ipv6network')
     )
-    return core_schema.function_plain_schema(
-        _validators.ip_v6_network_validator, serialization=core_schema.to_string_ser_schema(), metadata=metadata
+    return core_schema.lax_or_strict_schema(
+        lax_schema=core_schema.function_plain_schema(_validators.ip_v6_network_validator, metadata=metadata),
```

Comment:
Should the `metadata` (which is currently just used for JSON schema generation) be living on the parent `lax_or_strict_schema` rather than just the `lax` schema?

Reply:
Turns out, that doesn't work. I think we'd need to make some changes to our json schema generation helper functions, but that would greatly increase the scope of this PR, so I'm going to leave it as-is.

Reply:
okay, I'll be happy to take care of that after merging this